DP Mystery Map/Assets/Scripts/PlayerController.cs

```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
                 Player.FacingDirection = _walkDirection;
             }
 
+            //Update the idleFacing position per direction
+            switch (Player.FacingDirection)
+            {
+                case Direction.Up:
+                    playerAnimator.SetFloat("faceIdleX", 0);
+                    playerAnimator.SetFloat("faceIdleY", -1);
+                    break;
+                case Direction.Down:
+                    playerAnimator.SetFloat("faceIdleX", 0);
+                    playerAnimator.SetFloat("faceIdleY", 1);
+                    break;
+                case Direction.Left:
+                    playerAnimator.SetFloat("faceIdleX", -1);
+                    playerAnimator.SetFloat("faceIdleY", 0);
+                    break;
+                case Direction.Right:
+                    playerAnimator.SetFloat("faceIdleX", 1);
+                    playerAnimator.SetFloat("faceIdleY", 0);
+                    break;
+            }
+
             CheckKeys();
         }
         //else check for key being pressed down
         else
         {
             CheckKeys();
         }
+
+        //Handle direction faced while moving and idle
+            if (_isPlayerMoving == false)
+            {
+                playerAnimator.SetFloat("Horizontal", playerMovementController.x);
+                playerAnimator.SetFloat("Vertical", playerMovementController.y);
+                playerAnimator.SetFloat("Speed", playerMovementController.sqrMagnitude);
+            }
+            else
+            {
+            //Enable running animation if shift is held
+            if (Input.GetKey(KeyCode.LeftShift) && ((Input.GetKey(Player.MoveUpKey)) ||
```

Comment:
David this task wasn't assigned to you in the production plan. Also, this introduces a bug where the player can change their walk speed in the middle of a step.